### DISCUSSION TITLE
3.1.x release discussion

### DISCUSSION THREAD

**Discussion by Andyz26** [Netflix/mantis#684] [General]
hi @kmg-stripe @mcowgill-stripe @codyrioux I've merged the [zk leader PR](https://github.com/Netflix/mantis/pull/682). Given the scope of the change I am going to bump the version from 3.0.x to 3.1.x and the current rc build submitted on tag "v3.1.0-rc.1". Is RC build sufficient for your validation? Let me know if you want to start the official release here.

**Comment by kmg-stripe** [comment]
thanks @Andyz26 !  we are in the process of setting up a mirror for RC builds, so that should be sufficient for out testing.  i am hoping the mirror is up today.  i'll update here when we are done testing.

**Comment by kmg-stripe** [comment]
@Andyz26 Quick update.  We did some testing last week and are continuing to do more this week.  We did find a few bugs, so will likely have a few follow-up PRs (nothing major yet).  Hopefully in the next few days.

**Comment by kmg-stripe** [comment]
@Andyz26 We just merged a few small fixes.  I think we are close to being able to cut a release.  We have been running the 3.1.0-RC build for about a week and only found the few small issues.

A few requests:

1. Can we get the rights to approve a RC build?
2. Could @sarahwada-stripe get merge access?

Thanks!

**Comment by codyrioux** [comment]
+1 to getting Sarah merge access!

**Comment by Andyz26** [comment]
I've added @sarahwada-stripe to the collab users.
@kmg-stripe I don't see any other approval permission set on rc build. What happens if you push an rc tag e.g. 
[v3.1.0-rc.1](https://github.com/Netflix/mantis/tree/v3.1.0-rc.1) to remote? This should auto trigger the RC build.

**Reply by kmg-stripe** [reply]
@Andyz26 Nice!  Pushing the tag worked!  I just triggered `3.1.0-rc.2`

**Comment by kmg-stripe** [comment]
Hey @Andyz26 ,

I think we are getting close to being able to create the `3.1.x` release.  We have been trying to create an RC build, but keep getting  what appears to be Maven errors for some plugins: https://github.com/Netflix/mantis/actions/runs/9980799367/job/27618485399

@codyrioux was thinking the Maven repo may be out of sync?

**Comment by codyrioux** [comment]
Hey!

Replying from my phone. It’ll be a few hours before I’m online yet (need to
drop my kid off at daycare). As best I can tell from the errors I was
seeing last night some of Nebula’s transitive dependencies appear to be
missing.

**Comment by Andyz26** [comment]
yeah, I checked internally, and some old deps got removed on the Maven repo due to vulnerability reports. The recommendation is to upgrade to a newer release on the nebula plugin and I am testing it (running into some build issues right now).

**Reply by Andyz26** [reply]
testing a fix here: https://github.com/Netflix/mantis/pull/693 but currently blocked by the github workflow outage.

**Reply by Andyz26** [reply]
@kmg-stripe @codyrioux this should be fixed now.

**Reply by kmg-stripe** [reply]
awesome!  thanks so much @Andyz26!  i am going to cut another RC now.
